inst/markdown/compareScenarios2/cs2_02_macro.Rmd

```diff
@@ -115,6 +115,104 @@ vars %>%
   walk(showLinePlots, data = dPrice, scale = "fixed")
 ``` 
 
+### FE LCOE Prices Transport
+
+```{r FE LCOE Prices Transport}
+
+groups <- c(
+  "Price|Final Energy|Transport|Liquids|Petroleum",
+  "Price|Final Energy|Transport|Liquids|Diesel",
+  "Price|Final Energy|Transport|Liquids|Biomass",
+  "Price|Final Energy|Transport|Liquids|Efuel",
+  "Price|Final Energy|Transport|Gases|Natural Gas",
+  "Price|Final Energy|Transport|Gases|Efuel",
+  "Price|Final Energy|Transport|Electricity",
+  "Price|Final Energy|Transport|Hydrogen"
+)
+
+for (g in groups) {
+  
+  items <- c(
+    paste0(g, "|Fuel Cost"),
+    paste0(g, "|Transport and Distribution"), 
+    paste0(g, "|Carbon Price Component"), 
+    paste0(g, "|Other Taxes") 
+  )
+  
+  tot <- paste0(g, "|Total LCOE")
+  
+  showAreaAndBarPlots(dPrice, intersect(items, unique(dPrice$variable)), tot)
```

Comment:
It is not necessary to do a "safety-intersect" like that, as `showAreaAndBarPlots()` essentially calls `filter(dPrice, filter(variable %in% items))` and will ignore missing variables for the plot. But it will print a warning message telling you which variables are missing.

Reply:
I actually removed it, because while it was not crashing, the variables not used were still showing up in the legend, which i found confusing.

Reply:
oh, ok, then that is a bug in `showAreaAndBarPlots()`